packages/aleo-gateway/src/weighted_signers.rs

```diff
@@ -3,21 +3,33 @@ use cosmwasm_std::Uint128;
 use error_stack::Report;
 use multisig::key::PublicKey;
 use multisig::verifier_set::VerifierSet;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
 
 use crate::weighted_signer::WeightedSigner;
 use crate::{AleoValue, Error};
 
-#[derive(Debug, Clone)]
-pub struct WeightedSigners {
-    signers: Vec<WeightedSigner>, // TODO: [WeightedSigner; 32],
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct WeightedSigners<const GROUP_SIZE: usize = 2, const GROUPS: usize = 2> {
+    #[serde_as(as = "[[_; GROUP_SIZE]; GROUPS]")]
+    pub signers: [[WeightedSigner; GROUP_SIZE]; GROUPS],
     threshold: Uint128,
-    nonce: [u64; 4],
+    // nonce: [u64; 4], // TODO: this should be included before going to main net
 }
 
-impl TryFrom<&VerifierSet> for WeightedSigners {
+impl<const GROUP_SIZE: usize, const GROUPS: usize> TryFrom<&VerifierSet>
+    for WeightedSigners<GROUP_SIZE, GROUPS>
+{
```

Comment:
Please refer to me other comment over `proof.rs` where I share some ideas about this implementation, specifically about the strategy associated with the collection type we decide to use here.